### PR TITLE
Segments implemented

### LINKED ISSUE #533: Add Segments in Cutter
See Add Segments in Cutter
https://github.com/radare/radare2/pull/10288#event-1672464820 See iSSj

### PR DESCRIPTION
**Pull request by Vane11ope**
Segments is implemented as the way Sections is, Screenshot attached.

fix #533
segments

### CODE REVIEW

**Comment by Vane11ope** [comment]
https://github.com/radareorg/cutter/issues/533

File: src/widgets/SegmentsWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    case Qt::DisplayRole:
+        switch (segment) {
+        case SegmentsModel::NameColumn:
+            return tr("Name");
+        case SegmentsModel::SizeColumn:
+            return tr("Size");
+        case SegmentsModel::AddressColumn:
+            return tr("Address");
+        case SegmentsModel::EndAddressColumn:
+            return tr("EndAddress");
```
Comment (ITAYC0HEN) [inline src/widgets/SegmentsWidget.cpp]: Please change to "End Address"

File: src/widgets/SegmentsWidget.cpp [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        case SegmentsModel::PermColumn:
+            return tr("Perms");
```
Comment (ITAYC0HEN) [inline src/widgets/SegmentsWidget.cpp]: Please change to "Permissions"

**Comment by ITAYC0HEN** [comment]
Looks good to me :)
One last thing before merge, please implement the Filter option in order to search the segments by a name (Ctrl+F).

Requirements:
 - not visible by default
 - upon using Ctrl+F, the Search\Filter searchbox will appear and be focused 

You can see how it is implemented in other Widgets (for example, the flags widget)

**Comment by Vane11ope** [comment]
done with Filter

**Pull request merged**